articles/implementing-flask-wtform/index.md

```diff
@@ -0,0 +1,407 @@
+
+In your register route function, we check the request method to be a POST request, then in the if block, we get the form data sent by using the form member variable of the request object, then pass in the name value of each field in the HTML forms as a key to store the values.
+
+```python
+@app.route("/register", methods=["GET", "POST"])
+def register():
+   #check the request method to ensure the handling of POST request only
+   if request.method == "POST":
+       #store the form value
+       username = request.form["username"]
+       email = request.form["email"]
+       password = request.form["password"]
+ 
+   return render_template('register.html')
+```
+
+Your register view function should have a similar code. Recall that the value passed to the `request.form` method is our HTML form's name value. 
+
+The form data are handled adequately. The next process is to return the data to the webpage rather than redirecting the user to the registration form page.
+
+To return the data to the webpage, add the following line of code:
+```python
+@app.route("/register", methods=["GET", "POST"])
+def register():
+   #check the request method to ensure the handling of POST request only
+   if request.method == "POST":
+       #store the form value
+       username = request.form["username"]
+       email = request.form["email"]
+       password = request.form["password"]
+ 
+       return username + " <br/> " + email
+   return render_template('register.html')
+```
+By adding the return statement inside the request method block, the web app will redirect the user to a blank page and display their username data. Note that you need to restart your server for the code changes to take effect.
+
+If you’ve followed the instructions accordingly, then you should be able to handle a form using the request object. Now, let’s proceed to the other method of form creation and handling.
+
+# Form Creation With Flask-WTForms
+Flask WTForms is a library that makes form handling easy and structured. It also ensures the effective handling of form rendering, validation, and security.
+
+To build forms with this approach, you start by creating a new file in our app directory and name it `forms.py`. This file will contain all the application forms. Let’s create a new contact.html template, then the route with path `/contact` to handle the form logic, then render the contact template.
+
+In the implementation of our flask form, we need to install the Flask-WTForms first. 
+
+```bash
+$ pip install flask-wtf
+```
+Next, open the `forms.py` file created earlier. Here, we import the Flask form
+```python
+from flask_wtf import FlaskForm
+```
+The subsequent step is to create a contact form. To do this, we initialize a class that inherits the FlaskForm object. Your code should look similar to this.
```

Comment:
`FlaskForm` object. Whenever you refer to filenames, variables, or classes, make sure to use the proper formatting. Change this throughout the article.